yarn-project/prover-node/src/job/epoch-proving-job.ts

```diff
@@ -207,6 +212,36 @@ export class EpochProvingJob implements Traceable {
     }
   }
 
+  /**
+   * Kicks off a running promise that queries the archiver for the set of L2 blocks of the current epoch.
+   * If those changed, stops the proving job with a `rerun` state, so the node re-enqueues it.
+   */
+  private async scheduleEpochCheck() {
+    const intervalMs = Math.ceil((await this.l2BlockSource.getL1Constants()).ethereumSlotDuration / 2) * 1000;
+    this.epochCheckPromise = new RunningPromise(
+      async () => {
+        const blocks = await this.l2BlockSource.getBlockHeadersForEpoch(this.epochNumber);
+        const blockHashes = await Promise.all(blocks.map(block => block.hash()));
+        const thisBlockHashes = await Promise.all(this.blocks.map(block => block.hash()));
```

Comment:
can these 2 be combined in a `Promise.all`?

Reply:
It could, but it doesn't make much of a difference to be honest, since the only async part is initializing bb's singleton. After that everything happens synchronously.